### PR TITLE
Allow UPS Label Purchase with no payment method

### PR DESCRIPTION
**Pull request by bborman22**
**Summary**

Since UPS labels are charged to the user's connected UPS account there is no need for them to have a payment method on file with WCS to purchase a UPS label. This update adds the `carrierId` to the selected rate data so that it can be checked if a label purchase is allowed with no payment method. Currently the only `carrierId` that does not require a payment method is UPS. 

**How to Test**

1. With WC and WCS set up on a site and UPS enabled.
2. Connect a UPS account.
3. Remove any payment method attached to the account.
4. Go to print a shipping label for an order that would provide UPS rates.
5. Select a USPS rate and see that the sidebar prompts to add a credit card.
6. Select a UPS rate and see that the label size dropdown and buy label button are present.
7. Click Buy Label for the UPS rate and confirm that the label was purchased.

Additional Test: Follow the above steps with an order shipment broken into multiple packages. Confirm that if UPS is selected for all packages, you are able to buy the labels. If a non-UPS carrier is selected for any of the packages it should prompt the user to add a credit card.

### CODE REVIEW

File: client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/purchase-section/index.js [hunk 1]
```diff
@@ -39,6 +40,16 @@ const PurchaseSection = props => {
 	} = props;
 	const purchaseBusy = form.isSubmitting && ! form.needsPrintConfirmation;
 	const hasSelectedRate = hasSelectedRates( form.rates );
+	let labelRequiresPaymentMethod = false;
+	if( hasSelectedRate ) {
+		forEach( form.rates.values , ( label ) => {
```
Comment (Ferdev) [inline client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/purchase-section/index.js:45]: hey @bborman22 , what do you think about replacing this `forEach` with its ES6 version, `for..of`? That way we can reduce to the minimum the amount of lodash imports. I've been trying to do it lately as much as possible (reduce lodash dependencies, I mean).

File: client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/sidebar.js [hunk 2]
```diff
@@ -57,13 +59,24 @@ export const Sidebar = props => {
 	};
 	const onPaperSizeChange = value => props.updatePaperSize( orderId, siteId, value );
 
+	const hasSelectedRate = hasSelectedRates( form.rates );
+	let labelRequiresPaymentMethod = false;
+	if( hasSelectedRate ) {
+		forEach( form.rates.values , ( label ) => {
+			if( 'ups' !== label.carrierId ) {
```
Comment (Ferdev) [inline client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/sidebar.js:66]: I think this condition can be merged with the variable declaration above.

**Review by Ferdev — APPROVED** [review]
This looks really good! :rocket: Just a couple of minor comments.

File: client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/purchase-section/index.js (hunk 1 above)
Comment (bborman22) [reply client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/purchase-section/index.js:45]: Yeah I agree, I'll look to convert this to ES6 to remove the dependency.

File: client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/sidebar.js (hunk 2 above)
Comment (bborman22) [reply client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/sidebar.js:66]: Sorry for my confusion, would you be able to expand on this? I was definitely looking to condense this bit here if possible.

Comment (bborman22) [reply client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/sidebar.js]: I think I got this cleaned up and put the conditions right into the assignment now. If you wouldn't mind taking a look at the update and let me know if that makes sense to do it that way?

File: client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/purchase-section/index.js [hunk 3]
```diff
@@ -40,14 +39,9 @@ export const PurchaseSection = props => {
...
 	const hasSelectedRate = hasSelectedRates( form.rates );
-	let labelRequiresPaymentMethod = false;
-	if( hasSelectedRate ) {
-		forEach( form.rates.values , ( label ) => {
-			if( 'ups' !== label.carrierId ) {
-				labelRequiresPaymentMethod = true;
-			}
-		});
-	}
+	const labelRequiresPaymentMethod = hasSelectedRate && Object.values( form.rates.values ).some( ( label ) => {
```
Comment (Ferdev) [inline client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/purchase-section/index.js:42]: :heart_eyes:

File: client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/sidebar.js (hunk 2 above)
Comment (Ferdev) [reply client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/sidebar.js]: yeah, sorry Brian, my explanation was very poor. That was exactly what I meant :smile:

**Pull request merged**